libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/NavigationCameraTransitionListener.java

```diff
@@ -0,0 +1,22 @@
+package com.mapbox.services.android.navigation.ui.v5.camera;
+
+import com.mapbox.mapboxsdk.location.OnLocationCameraTransitionListener;
+
+class NavigationCameraTransitionListener implements OnLocationCameraTransitionListener {
+
+  private final NavigationCamera camera;
+
+  NavigationCameraTransitionListener(NavigationCamera camera) {
+    this.camera = camera;
+  }
+
+  @Override
+  public void onLocationCameraTransitionFinished(int cameraMode) {
+    camera.updateTransitionListenersFinished();
+  }
+
+  @Override
+  public void onLocationCameraTransitionCanceled(int cameraMode) {
+    camera.updateTransitionListenersCancelled();
```

Comment:
The `cameraMode` needs to be pushed to the internal listener and then pushed to the public listeners, otherwise, once canceled, the `OnTrackingModeTransitionListener#onTransitionCancelled(@NavigationCamera.TrackingMode int trackingMode)` is going to deliver `trackingMode` that _canceled_ this transition (and is now transitioning itself) rather than the one that scheduled this transition, because the `NavigationCamera#trackingCameraMode` is already going to be overwritten.

Reply:
And since those callbacks are posted to the back of the message queue in the Maps SDK, it'd apply the same logic to the `camera.updateTransitionListenersFinished();` above.